src/lib/AccountFinder.ts

```diff
@@ -0,0 +1,58 @@
+import { AccountInfo } from '@/lib/AccountInfo';
+import { NetworkClient } from '@nimiq/network-client';
+import { WalletStore } from '@/lib/WalletStore';
+
+export default class AccountFinder {
+    public static async findAccounts(
+        deriveAccounts: (startIndex: number, count: number) => Promise<Array<{ address: string, keyPath: string }>>,
+        deriveWalletId: (accounts: Array<{ address: string, keyPath: string, balance: number }>) => Promise<string>,
+        defaultLabel: string = 'Standard Account',
+        onIntermediateResult?: (accounts: Array<{ address: string, keyPath: string, balance: number }>) => void,
+    ): Promise<AccountInfo[]> {
+        await NetworkClient.Instance.init(); // initialize NetworkClient if not initialized yet
+        const foundAccounts: Array<{ address: string, keyPath: string, balance: number }> = [];
+        let nextStartIndex = 0;
+        let derivedAccountsPromise = deriveAccounts(nextStartIndex, AccountFinder.MAX_ALLOWED_GAP);
+        nextStartIndex += AccountFinder.MAX_ALLOWED_GAP;
+        while (true) {
+            const derivedAccounts = await derivedAccountsPromise;
+            if (derivedAccounts.length !== AccountFinder.MAX_ALLOWED_GAP) {
+                throw new Error('\'deriveAccounts\' returned wrong number of accounts');
```

Comment:
Can we maybe handle this more gracefully instead of throwing, or is this an unexpected error that we need to know about if it happens?

Reply:
It's an unexpected error that only get's thrown if we messed up in the code.
The `deriveAccounts` method knows via a parameter how may accounts it should derive.
This check could theoretically also be removed.

Reply:
Ok, then if we see this in Sentry, we know it broke a users login flow.

Reply:
On the other hand, does this actually break anything if less accounts are returned? I don't think so, because you are using the `derivedAccounts.length` to advance the index anyway. In that case you can just remove the check and continue regularily.

Reply:
All right, let's see if it comes up in the testnet.